### PR TITLE
[GPII-4328] Move alerts and lbm to the owner modules

### PR DESCRIPTION
**Pull request by amatas**
This PR moves the alerts policies and the lbms to the owner modules. Also some alerts depend on the flowmanager to be ready, in order to avoid noise when the cluster is deployed for the first time.

### CODE REVIEW

File: gcp/live/dev/k8s/gpii/flowmanager/terraform.tfvars [hunk 1]
```diff
@@ -19,10 +20,14 @@ terragrunt = {
 
 # ↓ Module configuration (empty means all default)
 
-cert_issuer_name     = "letsencrypt-staging"
+cert_issuer_name = "letsencrypt-staging"
```
Comment (natarajaya) [inline gcp/live/dev/k8s/gpii/flowmanager/terraform.tfvars]: These changes are probably accidental?

File: gcp/modules/backup-exporter/alert_backup_exporter_errors.tf [hunk 2]
```diff
@@ -4,7 +4,7 @@ resource "google_monitoring_alert_policy" "backup_exporter_errors" {
 
   conditions {
     condition_threshold {
-      filter          = "metric.type=\"logging.googleapis.com/user/backup_exporter.error\" resource.type=\"k8s_container\""
+      filter          = "metric.type=\"${google_logging_metric.backup_exporter_error.name}\" resource.type=\"k8s_container\""
```
Comment (natarajaya) [inline gcp/modules/backup-exporter/alert_backup_exporter_errors.tf]: I like that we can reference these by name!

File: gcp/modules/gpii-flowmanager/alert_cluster_resource_allocatable_utilization.tf [hunk 3]
```diff
@@ -1,7 +1,7 @@
 resource "google_monitoring_alert_policy" "cluster_resource_allocatable_utilization" {
+  depends_on   = ["null_resource.wait_for_lbms"]
```
Comment (natarajaya) [inline gcp/modules/gpii-flowmanager/alert_cluster_resource_allocatable_utilization.tf]: I don't think that cluster-wide policy belongs to flowmanager module. Maybe move cluster-wide alerts to cluster module?

**Comment by natarajaya** [comment]
Thanks @amatas , nice improvements!

I have only suggestion about cluster-wide alert policies (and other resources), otherwise LGTM.

File: gcp/modules/gpii-flowmanager/alert_cluster_resource_allocatable_utilization.tf (hunk 3 above)
Comment (amatas) [reply gcp/modules/gpii-flowmanager/alert_cluster_resource_allocatable_utilization.tf]: I know, the fact is that the flowmanager is the last package to be deployed and I wanted to deploy the alerts after all the resources of the cluster are available. That is why I set flowmanager module as a dependency for this and other alerts.

File: gcp/live/dev/k8s/gpii/flowmanager/terraform.tfvars (hunk 1 above)
Comment (amatas) [reply gcp/live/dev/k8s/gpii/flowmanager/terraform.tfvars]: I think that this was the vscode formatting..

File: gcp/modules/gpii-flowmanager/alert_cluster_resource_allocatable_utilization.tf (hunk 3 above)
Comment (natarajaya) [reply gcp/modules/gpii-flowmanager/alert_cluster_resource_allocatable_utilization.tf]: Got it, but I don't think they belong to flowmanager module anyway... Maybe keep them in separate module that deploys last if this is needed?

Comment (stepanstipl) [reply gcp/modules/gpii-flowmanager/alert_cluster_resource_allocatable_utilization.tf]: 👍 , perhaps not all alerts need to go after - would this particular alert, for example, cause an issue? If not, I think the logical place might be the cluster module

File: gcp/modules/gpii-flowmanager/alert_dns_modify.tf [hunk 4]
```diff
@@ -26,6 +26,8 @@ resource "google_monitoring_alert_policy" "dns_modify" {
     mime_type = "text/markdown"
   }
 
-  notification_channels = ["${google_monitoring_notification_channel.email.name}", "${google_monitoring_notification_channel.alerts_slack.*.name}"]
-  enabled               = "true"
+  notification_channels = ["${data.terraform_remote_state.alert_notification_channel.slack_notification_channel}", "${data.terraform_remote_state.alert_notification_channel.mail_notification_channel}"]
+
+  # Disabled on ephemeral clusters to avoid noise on recreation
+  enabled = "${(var.env == "prd" || var.env == "stg") ? "true" : "false"}"
```
[the remark this answers is not in the document]
Comment (amatas) [reply gcp/modules/gpii-flowmanager/alert_dns_modify.tf]: The alert is able to detect the modification of the DNS made at the ingress deployment:

imagen

Even when the alert was created minutes after of such modification.

[the remark this answers is not in the document]
Comment (amatas) [reply gcp/modules/gpii-flowmanager/alert_dns_modify.tf]: I have tried changing the aggregation time, just to see if we can shift the triggering of the alert without success.

**Comment by amatas** [comment]
as we agreed in our last standup meeting, the alerts that must be created at the end of the deployment are in an isolated module called "late-alerts"

**Pull request merged**